analyzers/tests/SonarAnalyzer.Test/TestCases/ClassShouldNotBeEmpty.cs

```diff
@@ -141,29 +141,58 @@ class Message { }                            // Compliant, ignored because of th
 
     class AssemblyDoc { }                        // Compliant, used for DefaultDocumentation tool
     class NamespaceDoc { }                       // compliant, used for DefaultDocumentation tool
+}
 
-    class BaseWithProtectedConstructor
+namespace ConstructorAccessibility
+{
+    public class PublicClassWithPublicConstructor
     {
-        protected BaseWithProtectedConstructor() { }
+        public PublicClassWithPublicConstructor() { }
     }
 
-    class WidensConstructorVisibility1 : BaseWithProtectedConstructor { }           // Compliant
+    public class PublicClassWithInternalConstructor
+    {
+        internal PublicClassWithInternalConstructor() { }
+    }
 
-    class BaseWithInternalConstructor
+    public class PublicClassWithProtectedConstructor
     {
-        internal BaseWithInternalConstructor() { }
+        protected PublicClassWithProtectedConstructor() { }
     }
 
-    class WidensConstructorVisibility2 : BaseWithInternalConstructor { }            // Compliant
+    internal class InternalClassWithPublicConstructor
+    {
+        public InternalClassWithPublicConstructor() { }
+    }
 
-    class BaseWithPublicConstructor
+    internal class InternalClassWithInternalConstructor
     {
-        public BaseWithPublicConstructor() { }
+        internal InternalClassWithInternalConstructor() { }
     }
 
-    class ConstructorIsPublicAlready1 : BaseWithPublicConstructor { }               // Noncompliant
+    internal class InternalClassWithProtectedConstructor
+    {
+        protected InternalClassWithProtectedConstructor() { }
+    }
+
+    public class ConstructorIsPublicAlready1 : PublicClassWithPublicConstructor { }             // Noncompliant
+    public class WidensConstructorAccessibility1 : PublicClassWithInternalConstructor { }       // Compliant
+    public class WidensConstructorAccessibility2 : PublicClassWithProtectedConstructor { }      // Compliant
 
-    class BaseWithDefaultConstructor { }                                            // Noncompliant
+    internal class ConstructorIsPublicAlready2 : PublicClassWithPublicConstructor { }           // Noncompliant
+    internal class ConstructorIsInternalAlready1 : PublicClassWithInternalConstructor { }       // Noncompliant
+    internal class WidensConstructorAccessibility3 : PublicClassWithProtectedConstructor { }    // Compliant
+    internal class ConstructorIsInternalAlready2 : InternalClassWithPublicConstructor { }       // Noncompliant
+    internal class ConstructorIsInternalAlready3 : InternalClassWithInternalConstructor { }     // Noncompliant
+    internal class WidensConstructorAccessibility4 : InternalClassWithProtectedConstructor { }  // Compliant
```

Comment:
What about adding a few more TC with `protected class` and various combinations?